src/parser/Parser.ts

```diff
@@ -1785,13 +1789,15 @@ export class Parser {
                     left.closingSquare
                 );
             } else if (isDottedGetExpression(left)) {
-                return new DottedSetStatement(
+                const dottedSetStmt = new DottedSetStatement(
                     left.obj,
                     left.name,
                     operator.kind === TokenKind.Equal
                         ? right
                         : new BinaryExpression(left, operator, right)
                 );
+                this._references.dottedSetStatements.push(dottedSetStmt);
```

Comment:
I think you'll collect dottedSets in anonymous functions attached potentially to AA objects and not to the top `m`. I think this risky and instead you should have a post-parsing walk of the statements of the top-level functions.

Reply:
Yeah, good Idea. I did not think of that